### DISCUSSION TITLE
(Proposal) Sunflower Dungeon

### DISCUSSION THREAD

**Discussion by Mbst36** [sunflower-land/sunflower-land#2285] [Ideas]
Special shout-out to Tourist#6002 on Discord for nudging me to submit this proposal.
Formatting on github is not like Word, I have attached my original Word doc for easier reading.

SFL Dungeon Outline
1. Summary
2. Potion Master and Blacksmith
3. Entrance to the Dungeon
4. Dungeon Lore
5. Dungeon Mechanics
6. Monster List
7. Dungeon Player Control
8. Degree of Difficulty Calculator
9. Skill Tree Progression

1.	Summary
      A.	PSA: I have zero coding experience; this is an outline of my dreams of a SFL dungeon. Although I cant code, I do work with economic models and create business contracts that protect my companies best interest and incentivize best practices. Said another way, I suck at coding, but this dungeon will provide some interesting mechanics.
      B.	Introduces a new gameplay mechanic, skill tree and utilization for Potion Master and Blacksmith
      C.	Introduces lore for the Necromancer, Moonseekers and Sunfloria.
      D.	Each dungeon level tier allows for traditional resource gathering and a new resource to use at the Blacksmith, Moon Dust
      E.	Introduces a resource sink for entering Dungeon and crafting potions
2.	Potion Master and Blacksmith
      A.	Potion Crafting
            i.	The Potion Master will make a Sunflower and Potato potion dungeon boost daily
              1.	500 Sunflowers
                  a.	Heals X
              2.	100 Potatoes
                  a.	Gains X damage dealt for X seconds
            ii.	The purpose of potion crafting is to provide a sink for low level resources in order to provide an advantage in the dungeon
      B.	Blacksmith crafting
            i.	Acquire Moon Dust to level up your weapon. All weapons will start on level 1 with base damage
              1.	(Hammer and Sword have additional damage boost?)
              2.	Level 2 Upgrade
                  a.	Increase damage by 1
                  b.	Increase health by 10
                  c.	Requirement: 500 Moondust
              3.	Level 3 Upgrade
                  a.	Increase damage by 3
                  b.	Increase health by 30
                  c.	Requirement: 1500 Moondust
              4.	Level 4 Upgrade
                  a.	Increase damage by 5
                  b.	Increase health by 50
                  c.	Requirement: 5000 Moondust
          ii.	The purpose of the upgrades is to have a goal for players to work toward. Moondust is acquired by defeating monsters in the dungeon. The level upgrades serves as a ceiling for advancing in the dungeon. Without a level 2 upgrade, it is extremely hard to proceed past the first tier of the dungeon. Without level 3, extremely hard to get past 2nd tier, and so on.
3.	Entrance into the Dungeon
        A.	Dungeon can only be entered 1 time per day
        B.	Each day a player must pay a nominal SFL fee and must complete either a “Work” quest or “Give” quest for the Goblin Gatekeeper to enter.
          i.	Sample “Work” Quests
              1.	Plant and give 100-200 Sunflowers
              2.	Plant and give 40-50 Potatoes
              3.	Plant and give 20-25 Pumpkins
        ii.	Sample “Give” Quests
              1.	Sell 600-700 Sunflowers
              2.	Sell 200-250 Potatoes
              3.	Sell 60-70 Pumpkins
        C.	The “Give” quests will provide another sink for resources and provide another avenue for newer players to gain SFL using the goblin balloon
4.	Dungeon Lore
        A.	1st time entering dungeon - A distressed goblin appears gasping for air as he explains he and his friends were lured to the dungeon by an undeniable scent. In the deep depths of the dungeon he was able to break his trance deep by sniffing his lucky sunflower. He came to his senses just in time to see his friends submerging into a pit of goo and re-emerging as moon seekers. Seated at the top of the pit was the Necromancer, holding the royalty of Sunfloria hostage. The royalty were bound and powerless to stop the goblins from walking into their doom.
        B.	After defeating the Necromancer – The Necromancer will narrowly escape through a portal with the Sunfloria royal in-tow, leaving behind the end-of-dungeon chest.
5.	Dungeon Mechanics
        A.	3 tiers of levels, 10 levels per tier  (Expand as necessary)
          i.	After each tier, players have the ability to leave the dungeon with the resources they have acquired
          ii.	After tier 1, if a player dies in the dungeon all the loot acquired is lost
          iii.	Each level is a 4x4 plot size to allow freedom of movement
      B.	Monster and Resource mechanics
          i.	Monsters – Each level has multiple waves of monsters
              1.	Each wave will have subsequently more monsters than the last
              2.	Each tier will have a set range of monsters spawned
        ii.	Resources – A 50/50 RNG roll will dictate if a resource spawns in a monster wave 5 seconds after the monsters have spawned. Resource will only be available to mine for a short period of time (10 seconds?)
              1.	Goal is to introduce strategy into gameplay
                  a.	Will you kill a few enemies before trying to mine resources?
      iii.	Environmental Obstacle – Provides additional difficulty and chaos to level progression, these can only damage the player
    C.	Spawning Zones
      i.	To start, there should be small radius around the player where nothing can spawn
      ii.	Monsters and resources will spawn randomly around the map
              1.	Some spawns it will be easier than others to snag a resource before the monster overwhelm you 
    D.	Tier 1
      i.	Entry level monsters (See Monster list below)
              1.	15-20 monsters spawned per level
      ii.	Resource: Stone
      iii.	Environmental Obstacle: None
      iv.	Entry Level Drop rate: 25% chance to drop 1 Moondust
     E.	Tier 2
      i.	Advanced and Entry Level Monsters
              1.	20-25 monsters spawned per level
                  a.	10-15 Entry Level
                  b.	10 Advanced Monsters
      ii.	Resource: Iron
      iii.	Environmental Obstacle: Lava spots appear
      iv.	Advanced Drop rate: 25% to drop 1-2 Moondust
     F.	Tier 3
      i.	Expert, Advanced and Entry Level
              1.	30-35 monsters spawned per level
                a.	10-15 Entry Level
                b.	10 Advanced
                c.	5-10 Expert
      ii.	Resource: Gold
      iii.	Environmental Obstacle: Lava river and boulders crashing
      iv.	Expert Drop rate: 25% to drop 1-3 Moondust
    G.	Necromancer Boss
      i.	Last level of the Dungeon will have a showdown with the necromancer
      ii.	Abilities – Raise the dead, Staff beam spell, Tractor Beam spell
      iii.	Defeating the Necromancer will give the player the dungeon loot box
              1.	Random resources and Moon Dust
6.	Monster List
    A.	Entry Level
      i.	Moonseekers – Basic attack
      ii.	Moonseeker Archers – Basic ranged attack
      iii.	Dazed Goblin – Basic attack
      iv.	Dazed Goblin Archer – Basic ranged attack
      v.	Cave Bat – Basic attack
      vi.	Crazed Chicken – Basic attack
     B.	Advanced
      i.	Moonseeker Brutes – Charged attack
      ii.	Dazed Goblin Brute – Charged attack
      iii.	Cobra – Basic attack
      iv.	Goo blob – Basic attack
      v.	Large Cave bat – Charge attack
    C.	Expert
      i.	Rock Monster – Charged attack
      1.	Splits into 3 smaller rock monsters once killed
      ii.	Mutant Brute Moonseeker – Charged Attack
      iii.	Mutant Chicken – Basic attack
      iv.	Parsnip Assassin – Assassin that’s a parsnip, be very afraid of the shuriken throws
7.	Dungeon Player Control
     A.	As previously mentioned, I have zero coding experience. Here is what I think would be somewhat feasible but allows for freedom of movement. I assume coding for PC will be easiest rather than both PC and mobile
     B.	Dungeon player controls – PC only
        i.	Up, down, left, right, jump and attack
              1.	Movement similar to bumpkin plaza with additional commands to jump and attack
8.	Degree of Difficulty Calculator (Need help)
    A.	*This system is important, but I have zero experience in brainstorming this concept. It will allow to easily scale the dungeon by plugging new monsters in without much tinkering*
    B.	        I think it would be similar to most MMORPG games, each monster has different animations but a similar base damage score
    C.	Each monster in their respective degree of difficulty will have the same damage and speed or damage score to simplify coding as much as possible
9.	Skill Tree Progression
    A.	Minor Speed Boost
    B.	        Minor Damage Boost
    C.	Increased drops of moondust and resources in-dungeon
[SFL Dungeon Proposal.docx](https://github.com/sunflower-land/sunflower-land/files/10830180/SFL.Dungeon.Proposal.docx)

**Comment by rasheed19871** [comment]
I like this idea but I think the developer or already very busy, and that seems tons of work but who knows maybe they like it and start working toward

**Reply by Mbst36** [reply]
Yes, this would probably be a project that starts development in 2024. Thanks for the feedback!
